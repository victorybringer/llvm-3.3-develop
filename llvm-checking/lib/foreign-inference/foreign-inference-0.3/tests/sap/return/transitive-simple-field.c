struct T {
  int *tp1;
  int *tp2;
};

struct S {
  struct T *t1;
  int *p2;
};

int *g(int x, struct T *t) {
  return t->tp2;
}

int *f(struct S *s) {
  return g(55, s->t1);
}
