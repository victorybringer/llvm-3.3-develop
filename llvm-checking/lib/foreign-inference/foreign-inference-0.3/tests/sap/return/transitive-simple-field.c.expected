fromList [ ("f", fromList [(0, "%struct.S*",[AccessDeref,AccessField 0,AccessDeref,AccessField 1])])
         , ("g", fromList [(1, "%struct.T*",[AccessDeref,AccessField 1])])
         ]
